python-package/lightgbm/sklearn.py

```diff
@@ -1298,10 +1296,7 @@ def _more_tags(self) -> Dict[str, Any]:
         return tags
 
     def __sklearn_tags__(self) -> "_sklearn_Tags":
-        tags = LGBMModel.__sklearn_tags__(self)
-        tags.estimator_type = "regressor"
-        tags.regressor_tags = _sklearn_RegressorTags(multi_label=False)
-        return tags
+        return super().__sklearn_tags__()
```

Comment:
This is necessary even though it doesn't do anything, because of this check in `scikit-learn`:

> E   TypeError: Estimator LGBMRegressor has defined either `_more_tags` or `_get_tags`, but not `__sklearn_tags__`. If you're customizing tags, and need to support multiple scikit-learn versions, you can implement both `__sklearn_tags__` and `_more_tags` or `_get_tags`. This change was introduced in scikit-learn=1.6

https://github.com/scikit-learn/scikit-learn/blob/fba028b07ed2b4e52dd3719dad0d990837bde28c/sklearn/utils/estimator_checks.py#L4459-L4465

Added in https://github.com/scikit-learn/scikit-learn/pull/30268